school/models.py

```diff
@@ -28,7 +35,48 @@ class Student(models.Model):
     promotion = models.ForeignKey(Promotion)
     graduate = models.BooleanField(default=False)
     user = models.OneToOneField(User)
-    artist = models.OneToOneField(Artist)
+    artist = models.OneToOneField(Artist, related_name='student')
 
     def __unicode__(self):
         return "%s (%s)" % (self.user, self.number)
+
+
+class StudentApplication(models.Model):
+    """
+    Fresnoy's School application procedure
+    """
+    artist = models.ForeignKey(Artist, related_name='student_application')
+
+    application_number = models.CharField(max_length=8, default=None, blank=True)
+
+    first_time = models.BooleanField(default=True, help_text="If the first time the Artist's applying")
+    last_application_year = models.PositiveSmallIntegerField(null=True, blank=True)
+    created_on = models.DateTimeField(auto_now_add=True)
+
+    remote_interview = models.BooleanField(default=False)
+    remote_interview_type = models.CharField(blank=True, max_length=50, help_text=_("Skype / Gtalk / FaceTime / AppearIn / Other"))
+    remote_interview_info = models.CharField(blank=True, max_length=50, help_text="ID / Number / ... ")
+
+    administrative_galleries = SortedManyToManyField(Gallery, blank=True, related_name='certificates')
+    artwork_galleries = SortedManyToManyField(Gallery, blank=True, related_name='artworks')
+
+    selected_for_interview = models.BooleanField(default=False, help_text="Is the candidat selected for the Interview")
+
+    def setApplicationNumber(self):
```

Comment:
the name is misleading: it doesn't set anything, but returns an application number. Also, see pep8 about method: lowercase and underscore. Last, make it an internal method. So, "_make_application_number" or "_generate_application_number" is better.